### PR TITLE
Move and include `@storybook/components` in standalone Storybook

### PR DESCRIPTION
**Pull request by JReinhold**
This PR continues the work from #19465 and adds the components to the mix.

## What I did

- Moved `@storybook/components` to `code/ui/components`.
- Simplified the main configuration as we didn't need the classic runtime after all.

Most notable is the fact that some stories in MDX currently doesn't work. The typography stories contained code fences which I had to remove, because `SyntaxHighlighter2` wasn't working, which hints that we might have issues with double imports when we have stories that include doc blocks. This makes sense and is somewhat expected when running a Storybook for Storybook. This is only an issue in mdx files though, and not even auto-generated docs, so I've disabled the offending features for now and might look into it later if I have more time.

## How to test

```
cd code
yarn storybook:ui
```

### CODE REVIEW

File: code/ui/components/src/Badge/Badge.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Badge } from './Badge';
```
Comment (JReinhold) [inline code/ui/components/src/Badge/Badge.stories.tsx:1]: The following stories in this PR have just been moved, and many of them have been converted from `storiesOf` to CSF3.

The diff shows completely new content, not really true.

File: code/ui/components/src/typography/DocumentWrapper.stories.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 
 import { DocumentWrapper } from './DocumentWrapper';
-import MarkdownSample from './DocumentFormattingSample.md';
+import MarkdownSample from './DocumentFormattingSample.mdx';
```
Comment (JReinhold) [inline code/ui/components/src/typography/DocumentWrapper.stories.tsx:4]: required to make Prettier format MDX2 comments correctly.

**Comment by JReinhold** [comment]
I'm very confused as to why this PR adds two new stories to **all** our sandbox Chromatics - a Primary and Secondary button in iframes..

@tmeasday any ideas?

**Comment by tmeasday** [comment]
> I'm very confused as to why this PR adds two new stories to all our sandbox Chromatics - a Primary and Secondary button in iframes..

> @tmeasday any ideas?

@JReinhold I think it's something unrelated because we briefly ran out of snapshots on Chromatic and a few builds on next were skipped. @shilman  accepted them all here: https://github.com/storybookjs/storybook/pull/19600, I suspect if you update from `next` you should be OK.

File: code/ui/.storybook/preview.jsx [hunk 3]
```diff
@@ -9,8 +9,8 @@ import {
   convert,
   styled,
   useTheme,
-} from '@storybook/theming';
-import { Symbols } from '@storybook/components';
+} from '../../lib/theming';
```
[the remark this answers is not in the document]
Comment (tmeasday) [reply code/ui/.storybook/preview.jsx:12]: 🤔 are there symlinks involved @JReinhold ? How is this SB installed?

File: code/ui/components/src/Badge/Badge.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+export default {
+  title: 'Basics/Badge',
```
Comment (tmeasday) [inline code/ui/components/src/Badge/Badge.stories.tsx:4]: We should move towards using `autotitle` for all our stories (not necessarily in this PR)

**Review by tmeasday — APPROVED** [review]
Looks great! 

One question: are we testing these SBs with Chromatic? We should set that up.

File: code/ui/components/src/Badge/Badge.stories.tsx (hunk 4 above)
Comment (JReinhold) [reply code/ui/components/src/Badge/Badge.stories.tsx:4]: As you probably noticed I did this in #19615 where possible.

**Comment by JReinhold** [comment]
> One question: are we testing these SBs with Chromatic? We should set that up.

I'm working on this as we speak, in a follow up PR.

File: code/ui/.storybook/preview.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tmeasday) [reply code/ui/.storybook/preview.jsx:12]: You could try running with `NODE_OPTIONS="--preserve-symlinks --preserve-symlinks-main"` and see if it helps.

[the remark this answers is not in the document]
Comment (JReinhold) [reply code/ui/.storybook/preview.jsx:12]: Holy s**t that worked, even fixed the issue with `SyntaxHighlighter2` being undefined which has driven me nuts.
You have to explain the logic behind this, this doesn't match at all with what I thought the problem was.

[the remark this answers is not in the document]
Comment (tmeasday) [reply code/ui/.storybook/preview.jsx:12]: I can explain what it does. I have no idea why it fixes the problem but I'm sure we can work that out together :)

**Pull request merged**